matter_server/server/device_controller.py

```diff
@@ -696,41 +693,70 @@ async def ping_node(self, node_id: int) -> NodePingResult:
-                Clusters.GeneralDiagnostics.Enums.InterfaceTypeEnum.kUnknownEnumValue,
-            ):
-                continue
+            clean_ip = ip_address.split("%")[0]
+            node_logger.debug("Pinging address %s", clean_ip)
+            result[clean_ip] = await ping_ip(ip_address, timeout)
 
-            # enumerate ipv4 and ipv6 addresses
-            for ipv4_address_hex in network_interface.IPv4Addresses:
-                ipv4_address = convert_ip_address(ipv4_address_hex)
-                tasks.append(_do_ping(ipv4_address))
-            for ipv6_address_hex in network_interface.IPv6Addresses:
-                ipv6_address = convert_ip_address(ipv6_address_hex, True)
-                tasks.append(_do_ping(ipv6_address))
+        ip_addresses = await self.get_node_ip_addresses(
+            node_id, prefer_cache=False, scoped=True
+        )
+        tasks = [_do_ping(x) for x in ip_addresses]
+        # TODO: replace this gather with a taskgroup once we bump our py version
         await asyncio.gather(*tasks)
+
+        # retrieve the currently connected/used address which is used
+        # by the sdk for communicating with the device
+        if TYPE_CHECKING:
+            assert self.chip_controller is not None
+        if sdk_result := await self._call_sdk(
+            self.chip_controller.GetAddressAndPort, nodeid=node_id
+        ):
+            active_address = sdk_result[0]
+            node_logger.info(
+                "The SDK is communicating with the device using %s", active_address
+            )
+            if active_address not in result and node.available:
+                # if the sdk is connected to a node, treat the address as pingable
+                result[active_address] = True
+
         return result
 
+    @api_command(APICommand.GET_NODE_IP_ADRESSES)
+    async def get_node_ip_addresses(
+        self, node_id: int, prefer_cache: bool = False, scoped: bool = False
+    ) -> list[str]:
+        """Return the currently known (scoped) IP-adress(es)."""
+        cached_info = self._last_known_ip_addresses.get(node_id, [])
+        if prefer_cache and cached_info:
+            return cached_info if scoped else [x.split("%")[0] for x in cached_info]
+        node = self._nodes.get(node_id)
+        if node is None:
+            raise NodeNotExists(
+                f"Node {node_id} does not exist or is not yet interviewed"
+            )
+        node_logger = LOGGER.getChild(f"[node {node_id}]")
+        # query mdns for all IP's
+        # ensure both fabric id and node id have 16 characters (prefix with zero's)
+        fabric_id_part = self.fabric_id_hex.upper()
+        while len(fabric_id_part) < 16:
+            fabric_id_part = "0" + fabric_id_part
+        node_id_part = f"{node_id:0x}"
+        while len(node_id_part) < 16:
+            node_id_part = "0" + node_id_part
+        mdns_name = f"{fabric_id_part}-{node_id_part}.{MDNS_TYPE_OPERATIONAL_NODE}"
```

Comment:
This can be simplified to:

```suggestion
        mdns_name = f"{self.compressed_fabric_id:0{16}X}-{node_id:0{16}X}.{MDNS_TYPE_OPERATIONAL_NODE}"
```

Reply:
I'll adjust it but I had some trouble with pytest choking on this syntax a while ago so I did it the manual way. I''' double check how it behaves now.